lib/profileCache.js

```diff
@@ -51,28 +51,27 @@ function getProfileCacheKey(req) {
 }
 
 
-module.exports = function profileCache(server, options, next) {
+module.exports = function profileCache(server, options) {
 
   // Fetch all available profile data given the scopes on the request,
   // using cached data if possible.
 
-  server.method('profileCache.get', (req, next) => {
-    batch(req, {
+  server.method('profileCache.get', (req) => {
+    return batch(req, {
       '/v1/_core_profile': true,
       '/v1/uid': true,
       '/v1/avatar': ['avatar', 'avatarDefault'],
       '/v1/display_name': true
     })
-    .then(result => {
-      // Only cache the result if we can produce a suitable cache key.
-      const ttl = getProfileCacheKey(req) ? undefined : 0;
-      return next(null, result, ttl);
-    })
-    .catch(next);
+      .then(result => {
```

Comment:
@vladikoff , I have fixed the server stop function but it still didn't work. This method seems to be failing.